kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RealRenderTester.kt

```diff
@@ -213,6 +213,29 @@ internal class RealRenderTester<PropsT, StateT, OutputT : Any, RenderingT>(
     return expected
   }
 
+  private inline fun <reified T : ExpectedWorker<*>> consumeExpectedWorker(
+    predicate: (T) -> Boolean,
+    description: () -> String
+  ): T? {
+    val matchedExpectations = expectations.filterIsInstance<T>()
+        .filter(predicate)
+    return when (matchedExpectations.size) {
+      1 -> {
```

Comment:
The fact that we had to break out a separate function for workers is a little weird. Why does it make sense to allow leniency for workers but not for workflows?

Reply:
Answered here: https://github.com/square/workflow/issues/959#issuecomment-631001737